tests/unit/amo/pages/TestUserProfile.js

```diff
@@ -106,488 +95,446 @@ describe(__filename, () => {
   });
 
   it('renders AddonsByAuthorsCard for extensions', () => {
     const userId = 123;
-    const { params, store } = signInUserWithUserId(userId);
+    const { params } = signInUserWithProps({ userId });
+    const user = getCurrentUser(store.getState().users);
 
-    const root = renderUserProfile({ params, store });
+    renderUserProfile({ params });
 
-    expect(root.find(AddonsByAuthorsCard).at(0)).toHaveProp(
-      'addonType',
-      ADDON_TYPE_EXTENSION,
-    );
-    expect(root.find(AddonsByAuthorsCard).at(0)).toHaveProp(
-      'pageParam',
-      'page_e',
-    );
-    expect(root.find(AddonsByAuthorsCard).at(0)).toHaveProp('paginate', true);
-    expect(root.find(AddonsByAuthorsCard).at(0)).toHaveProp(
-      'pathname',
-      `/user/${userId}/`,
-    );
+    expect(screen.getByText(`Extensions by ${user.name}`)).toBeInTheDocument();
+    const extensionCard = screen
+      .queryAllByClassName('AddonsByAuthorsCard')
+      .item(0);
+    expect(
+      within(extensionCard).queryByText('Previous'),
+    ).not.toBeInTheDocument();
+    expect(within(extensionCard).queryByText('Next')).not.toBeInTheDocument();
   });
 
   it('renders AddonsByAuthorsCard for themes', () => {
     const userId = 123;
-    const { params, store } = signInUserWithUserId(userId);
+    const { params } = signInUserWithProps({ userId });
+    const user = getCurrentUser(store.getState().users);
 
-    const root = renderUserProfile({ params, store });
+    renderUserProfile({ params });
 
-    expect(root.find(AddonsByAuthorsCard).at(1)).toHaveProp(
-      'addonType',
-      ADDON_TYPE_STATIC_THEME,
-    );
-    expect(root.find(AddonsByAuthorsCard).at(1)).toHaveProp(
-      'pageParam',
-      'page_t',
-    );
-    expect(root.find(AddonsByAuthorsCard).at(1)).toHaveProp('paginate', true);
-    expect(root.find(AddonsByAuthorsCard).at(1)).toHaveProp(
-      'pathname',
-      `/user/${userId}/`,
-    );
+    expect(screen.getByText(`Themes by ${user.name}`)).toBeInTheDocument();
+    const themeCard = screen.queryAllByClassName('AddonsByAuthorsCard').item(1);
+    expect(screen.queryByText(themeCard, 'Previous')).not.toBeInTheDocument();
+    expect(screen.queryByText(themeCard, 'Next')).not.toBeInTheDocument();
   });
 
   it('renders a not found page if the API request is a 404', () => {
-    const { store } = dispatchSignInActions();
-    const errorHandler = new ErrorHandler({
-      id: 'some-error-handler-id',
-      dispatch: store.dispatch,
-    });
-    errorHandler.handle(
-      createApiError({
+    const userId = 123;
+    createFailedErrorHandler({
+      error: createApiError({
         response: { status: 404 },
         apiURL: 'https://some/api/endpoint',
         jsonResponse: { message: 'not found' },
       }),
-    );
+      id: createErrorHandlerId({ userId }),
+      store,
+    });
 
-    const root = renderUserProfile({ errorHandler, store });
+    renderUserProfile({ params: { userId } });
```

Comment:
It may be in other places, but here we are creating the `errorhandler` with `userId` so don't we need to pass it?  Or I can create the `errorHandler` with what I know is the default for `renderUserProfile` (100) but that feels to me like magic functionality.